### PR TITLE
Add documentation for installing Serverless locally

### LINKED ISSUE #3677: Update docs to show how Serverless can be installed and used locally
# This is a Feature Proposal

## Description
Instead of globally installing Serverless, could we only globally install a CLI shim that checks the current directory's `node_modules` for Serverless? This way we can add a `devDependency` directly to our `package.json` and lock it from there rather than having to do it in `serverless.yml`. 

There should be little-to-no changes from a developer's perspective.

### PR DESCRIPTION
**Pull request by hassankhan**
## What did you implement:

Closes #3677

## How did you implement it:

Added documentation as per the discussion in the issue.

## How can we verify it:
By checking `docs/getting-started.md`.

## Todos:

- [X] Write tests
- [X] Write documentation
- [X] Fix linting errors
- [X] Make sure code coverage hasn't dropped
- [X] Provide verification config / commands / resources
- [X] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -12,13 +12,22 @@ menuItems:
...
 
-First things first, lets get the serverless framework installed on your machine.
+First things first, let's get the Serverless framework installed on your machine.
 
 ```bash
 # Installing the serverless cli
 npm install serverless -g
 ```
 
+## Installing Serverless in an existing project
```
Comment (HyperBrain) [inline docs/getting-started.md]: I think "Invoking Serverless locally from a project" should be the next adjacent section:

## Invoking Serverless locally from within a project

To execute the locally installed Serverless executable you have to reference the binary out of the node modules directory.

Example:
```
node ./node_modules/serverless/bin/serverless deploy
```

**Comment by hassankhan** [comment]
@pmuens I'm not against moving it, it's just in the issue I was told 'Getting Started' would be the best place. If everyone agrees that it should move to 'Version Pinning', then I'll do that. 

@HyperBrain Good shout, I'll update the PR this evening.

**Comment by pmuens** [comment]
@hassankhan thanks for updating it 👍 

Pinging @eahefnawy and @DavidWells in the issue as well so that feedback in the location can be provided.

**Comment by hassankhan** [comment]
Thanks @pmuens!!

**Comment by pmuens** [comment]
Thanks for getting this started @hassankhan 👍

**Pull request merged**